### PR TITLE
Show Location edit icon to any logged-in user

### PR DESCRIPTION
**Pull request by mo-nathan**
## Summary
- The name-adjacent edit icon on the Location show page was gated on `AbstractModel#can_edit?`, which for Location reduces to "creator or admin only". That didn't match `LocationsController` — edit/update only require login, so any logged-in user can actually edit a location.
- The Coordinates and Notes subpanels each rendered their own unconditional edit icon to work around the over-strict header gate, leaving the creator with three edit icons (header + two panels) and non-creators with two (panels only).
- Fix at the model layer: `Location#can_edit?` returns true for any logged-in user. Drop the workaround icons.

## Changes
- `app/models/location.rb` — override `can_edit?` to allow any logged-in user (matches controller policy). Destroy is still creator/admin-only via `can_destroy_location?`, unchanged.
- `app/views/controllers/locations/show/_coordinates.erb` — drop redundant edit icon.
- `app/views/controllers/locations/show/_notes.html.erb` — drop redundant edit icon; collapse heading back to a plain `:NOTES.l`.
- `test/models/location_test.rb` — add unit test covering creator / non-creator / nil-user cases for `can_edit?`.

## Test plan
- [x] `bin/rails test test/models/location_test.rb test/controllers/locations_controller_test.rb` — 97 tests, 904 assertions, 0 failures
- [x] `bundle exec rubocop` clean on touched Ruby files
- [x] Manual: visit `/locations/:id` as a non-creator logged-in user; confirm a single edit icon appears next to the location name and none in Coordinates/Notes panels. Click it; confirm the edit form loads.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/models/location.rb [hunk 1]
```diff
@@ -708,6 +708,13 @@ def destroyable?
       descriptions.empty?
   end
 
+  # Locations are a shared resource: any logged-in user can edit one.
+  # (LocationsController#edit/#update only requires login. Destroy is
+  # still owner/admin-only — see can_destroy_location?.)
+  def can_edit?(user)
+    user.present?
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/models/location.rb]: Applied — reshaped to `return false unless user` / `true` to match AbstractModel#can_edit? idiom. See 92f17cdfa.

File: app/views/controllers/locations/show/_coordinates.erb [hunk 2]
```diff
@@ -1,6 +1,4 @@
-<% heading_links = [
-  icon_link_to(*edit_location_tab(@location))
-]
+<% heading_links = []
 if @location.destroyable? && (in_admin_mode? || @location.user == @user)
   heading_links << destroy_button(target: @location, icon: :delete)
 end
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/views/controllers/locations/show/_coordinates.erb:4]: Applied — guarded `panel.with_heading_links` on `heading_links.any?`. See 92f17cdfa.
